phonon/reference.py

```diff
@@ -226,11 +226,14 @@ def dereference(self, callback=None, args=None, kwargs=None):
             if self.__process.id in pids:  # It won't be here if a dereference previously failed at the delete step.
                 del pids[self.__process.id]
 
-        # Check for failed processes
-        pids = self.remove_failed_processes(pids)
+        if self.force_expiry:
+            pids = {}
+        else:
```

Comment:
Lines 222 - 227 can be moved into the else statement.  Might make a marginal performance improvement since they are unnecessary when `self.force_expiry` is true.